src/movies.js

```diff
@@ -1,25 +1,85 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map(function (directorsObj) {
+    return directorsObj.director;
+  });
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  const stevenMovies = moviesArray.filter(function (movies) {
+    return movies.director === "Steven Spielberg" && movies.genre.includes("Drama");
+  });
+  return stevenMovies.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) {
+    return 0;
+  }
+  const calcAvg = moviesArray.reduce(function (acc, movies) {
+    return acc + movies.score;
```

Comment:
In order to check if the movie has a score we can create an if statement that check if the value score exist inside the movie, if it exist add it to acc if not only return the acc 
```suggestion
        if (movies.score) {
            return acc + movies.score;
        }
        return acc;
```